### DISCUSSION TITLE
Can I inspect a scope for all the dependencies resolved in the scope?

### DISCUSSION THREAD

**Discussion by RLittlesII** [dadhi/DryIoc#659] [Q&A]
I have an application that uses scoping.  We are trying to verify what is actually resolved from a given scope to better manage our scoping concerns.

 1. Is there any way to look at a scope and determine the dependencies resolved from it?
 2. Is there any way to compare two scopes and determine the delta resolved?
 3. Is there any way to get the entire dependency graph for the resolution of a scope?

```csharp

public interface IRootThing {} // Root Scope

public interface IScopedThing {} // Sub-Scope

public interface IScopedDependency {} // Dependency with a full graph of dependencies
```

**Comment by ChaseFlorell** [comment]
Another way to word this is, how can I discover everything that the container constructed inside a given scope?

```csharp
public record Foo(Bar bar, IResolverContext context);
public class Bar(Bazz bazz);
public class Bazz(Fizz fizz);
public class Fizz(Buzz buzz);
public class Buzz();

// a few moments later
var foo = resolverContext.Resolve<Foo>();
foo.context.PrintThingsInScope();
// Foo, Bar, Bazz, Fizz, Buzz
```

**Comment by dadhi** [comment]
HI @RLittlesII 

- Do you mean the object graph of dependencies of the specific registered services (say services registered with `Reuse.Scoped`)?
- All services resolved from the specific scope, after they were resolved or injected, and their object graphs?

Could you provide the complete registration and resolution example for the Foo service you've mentioned?

**Comment by dadhi** [comment]
I have added the missing method into `IScope` and here is the example of the possible usage https://github.com/dadhi/DryIoc/blob/master/test/DryIoc.IssuesTests/GHIssue659_Can_I_inspect_a_scope_for_all_the_dependencies_resolved_in_the_scope.cs

**Comment by RLittlesII** [comment]
@dadhi Thanks.  This gets us much closer to where we want to be.  Thank You so much for this!
